lib/post_scraper.rb

```diff
@@ -16,17 +16,17 @@ class PostScraper < Object
     'maggie_of_the_owls' => 'MaggieoftheOwls' # have both - and _ versions cause Dreamwidth supports both
   }
 
-  attr_reader :url
+  attr_accessor :url, :post, :html_doc
 
   def initialize(url, board_id=nil, section_id=nil, status=nil, threaded_import=false, console_import=false)
     @board_id = board_id || SANDBOX_ID
     @section_id = section_id
     @status = status || Post::STATUS_COMPLETE
-    url = url + (if url.include?('?') then '&view=flat' else '?view=flat' end) unless url.include?('view=flat')
+    url = url + (if url.include?('?') then '&view=flat' else '?view=flat' end) unless url.include?('view=flat') || threaded_import
     url = url + '&style=site' unless url.include?('style=site')
```

Comment:
This probably isn't safe any more; it assumes that you're guaranteed a ? because of the previous line. I'd move the if statement to be this block, and the view=flat to be the 'well you're guaranteed now" one.